### PR TITLE
 Rewrite mathematica parsing function and add tests

### PR DESCRIPTION
**Pull request by quasi-human**
After commiting #13529, there were also some problems found in mathematica parsing. For example, the curent version cannot parse -1/(2*Sqrt[2]). So I renewed the function. I also added several tests. Both all the conventional tests and newly added tests were confirmed to be passed.

### CODE REVIEW

File: sympy/parsing/mathematica.py [hunk 1]
```diff
@@ -1,67 +1,133 @@
...
+    )
+
+    rules1 = (
+        # 'Arc' to 'a'
+        (r'''
+                (               # group1
+                .*              # any characters
+                [^a-zA-Z]       # a single character except alphabet
+                )
+                Arc             # detect 'Arc'
```
Comment (Upabjojr) [inline sympy/parsing/mathematica.py]:
```
In[1]:= ArcSin[1/2]                                                             

        Pi
Out[1]= --
        6

In[2]:= Arcsin[1/2]                                                             

               1
Out[2]= Arcsin[-]
               2
```
Are you sure it is parsing correctly?

Comment (Upabjojr) [reply sympy/parsing/mathematica.py]: I mean, `Arcsin` is a wrong spelling, it shouldn't be parsed to `asin`.

Comment (quasi-human) [reply sympy/parsing/mathematica.py]: Thank you for pointing it out.
I supposed that all input were correct expression in Mathematica.
The current mathematica.py also parses 'Arcsin[1/2]' to 'pi/6'.

Your point is that an input-check function should be introduced, right?

Comment (smichr) [reply sympy/parsing/mathematica.py]: I'm in favor of committing this as is and letting that issue be a separate PR. Any objections?

Comment (Upabjojr) [reply sympy/parsing/mathematica.py]: No objections.

Comment (quasi-human) [reply sympy/parsing/mathematica.py]: Thanks for kind reviewing, everybody.

Comment (kenogo) [reply sympy/parsing/mathematica.py]: I'd like to start working on checking the input expressions if nobody else has started on that.

Comment (quasi-human) [reply sympy/parsing/mathematica.py]: Thank you, kengo! But, soon I will send new pull request for making further rigid pursing.
So just wait a little bit, please.
And I am sorry if you already started checking.

**Pull request merged**